lua/nvim-treesitter/query.lua

```diff
@@ -156,8 +156,8 @@ function M.iter_prepared_matches(query, qnode, bufnr, start_row, end_row)
       for id, node in pairs(match) do
         local name = query.captures[id] -- name of the capture in the query
         if name ~= nil then
-          local path = split(name)
-          insert_to_path(prepared_match, path, { node=node })
+          local path = split(name..'.node')
+          insert_to_path(prepared_match, path, node)
```

Comment:
Maybe move this local function (`insert_to_path`) to `utils` there's a function `get_at_path` which is basically the inverse.

Reply:
Get at path would just be a standard dereference operation, I don't think we need this function.